projects/packages/my-jetpack/_inc/components/welcome-banner/index.jsx

```diff
@@ -18,27 +18,53 @@ import styles from './style.module.scss';
 const WelcomeBanner = () => {
 	const { recordEvent } = useAnalytics();
 	const { isWelcomeBannerVisible, dismissWelcomeBanner } = useWelcomeBanner();
-	const { isRegistered, isUserConnected } = useConnection();
-	const navigateToConnectionPage = useMyJetpackNavigate( MyJetpackRoutes.Connection );
+	const { isRegistered, isUserConnected, handleRegisterSite, siteIsRegistering } = useConnection( {
+		skipUserConnection: true,
+	} );
 	const [ bannerVisible, setBannerVisible ] = useState( isWelcomeBannerVisible );
+	const { setNotice, resetNotice } = useContext( NoticeContext );
 	const shouldDisplayConnectionButton = ! isRegistered || ! isUserConnected;
+	const siteConnectionButtonLabel = __( 'Finish setting up Jetpack', 'jetpack-my-jetpack' );
 
 	useEffect( () => {
 		if ( bannerVisible ) {
 			recordEvent( 'jetpack_myjetpack_welcome_banner_view' );
 		}
 	}, [ bannerVisible, recordEvent ] );
 
-	const onDismissClick = useCallback( () => {
-		recordEvent( 'jetpack_myjetpack_welcome_banner_dismiss_click' );
-		setBannerVisible( false );
-		dismissWelcomeBanner();
-	}, [ recordEvent, dismissWelcomeBanner ] );
+	const onDismissClick = useCallback(
+		( isDismissFromConnectionButton = false ) => {
+			if ( ! isDismissFromConnectionButton ) {
+				recordEvent( 'jetpack_myjetpack_welcome_banner_dismiss_click' );
+			}
+			setBannerVisible( false );
+			dismissWelcomeBanner();
+		},
+		[ recordEvent, dismissWelcomeBanner ]
+	);
 
 	const onFinishConnectionClick = useCallback( () => {
 		recordEvent( 'jetpack_myjetpack_welcome_banner_finish_connection_click' );
-		navigateToConnectionPage();
-	}, [ recordEvent, navigateToConnectionPage ] );
+		handleRegisterSite().then( () => {
+			onDismissClick( true );
```

Comment:
WDYT about calling 
```
setBannerVisible( false );
dismissWelcomeBanner();
```

here instead of re-using the `onDismissClick` handler? The boolean parameter and conditional handling seems more complex.